### PR TITLE
changes to get mosart river model nuopc cap working with nuopc-cmeps

### PR DESCRIPTION
**Pull request by mvertens**
changes to get mosart river model nuopc cap working

This PR does several things:

1. It incorporates changes in nuopc-cmeps that are necessary to have mosart working with ctsm. 
   -  A new test for this has been added to the testlist - 
`SMS_Vnuopc_Ln10.f19_f19_mg17.I2000Clm50SpGs.cheyenne_intel.clm-nuopc_cap`
The results of this test when examining mosart history files are that they are round-off level different in comparing with 
`SMS_Vmct_Ln10.f19_f19_mg17.I2000Clm50SpGs.cheyenne_intel.clm-nuopc_cap`
2. seq_comm_mct.F90 has been removed from src/drivers/nuopc/shr - and esm.F90 has been simplified to account fo this. 
3. Externals that need to be changed  (everything is still at the head of the branches)
   - cam was slightly modified to account for the seq_comm_mct removal - the branch needs to be updated (nuopc_cap_tags/nuopc_cap_n28_cam6_1_008)
   - mosart needs to be updated (bf53c0b)

Test suite:
`./create_test --xml-testlist ../src/drivers/nuopc/cime_config/testdefs/testlist_drv.xml --xml-machine cheyenne --xml-category nuopc --walltime 00:30:00 --compare jan24 --baseline-root /glade/scratch/dunlap/BASELINES  `
Test baseline: /glade/scratch/dunlap/BASELINES/jan24
Test namelist changes: no
Test status: bit for bit except for 2 rof fields in X compsets 
Fixes: none
User interface changes?: no
Update gh-pages html (Y/N)?:N
Code review:

### CODE REVIEW

File: src/drivers/nuopc/cime_driver/esm.F90 [hunk 1]
```diff
@@ -327,8 +321,8 @@ subroutine SetRunSequence(driver, rc)
     !   file=__FILE__)) &
     !   return  ! bail out
 
-!    call pretty_print_nuopc_freeformat(runSeqFF, 'run sequence', rc=rc)
-!    if (shr_nuopc_methods_ChkErr(rc,__LINE__,u_FILE_u)) return
+    ! call pretty_print_nuopc_freeformat(runSeqFF, 'run sequence', rc=rc)
```
Comment (jedwards4b) [inline src/drivers/nuopc/cime_driver/esm.F90:324]: Why are these here if they are commented out?   Should they instead be in a 
#ifdef DEBUG block?

File: src/drivers/nuopc/cime_driver/esm.F90 [hunk 2]
```diff
@@ -865,18 +862,7 @@ subroutine InitAttributes(driver, mastertask, rc)
...
-       ! in med_infodata calls, reset here and the copied back into med_infodata - this is no longer the case
-       ! call shr_scam_checkSurface(scmlon, scmlat, &
-       !      OCNID(ens1), mpicom_OCNID,            &
-       !      lnd_present=lnd_present,              &
-       !      ocn_present=ocn_present,              &
-       !      ice_present=ice_present,              &
-       !      rof_present=rof_present,              &
-       !      flood_present=flood_present,          &
-       !      rofice_present=rofice_present)
+       ! TODO(mvertens, 2019-01-30): need to add signel column functionality
```
Comment (jedwards4b) [inline src/drivers/nuopc/cime_driver/esm.F90]: spelling :-)

File: src/drivers/nuopc/cime_driver/esm.F90 [hunk 3]
```diff
@@ -1072,33 +1071,45 @@ subroutine AddAttributes(gcomp, driver, config, compid, compname, inst_suffix, r
...
 
-    ! inst_name and inst_index are no longer required for cime internal components
+    !------
+    ! Add multi-instance specific attributes
+    !------
     call NUOPC_CompAttributeAdd(gcomp, attrList=(/'inst_name','inst_index'/), rc=rc)
     if (shr_nuopc_methods_ChkErr(rc,__LINE__,u_FILE_u)) return
 
-    call NUOPC_CompAttributeSet(gcomp, name='inst_name', value=trim(seq_comm_name(compid)), rc=rc)
+    ! add inst_name attribute (inst_name not required for cime internal components)
```
Comment (jedwards4b) [inline src/drivers/nuopc/cime_driver/esm.F90]: I don't think that you needed to put this (inst_name) back did you?   Can we look at it together?

File: src/drivers/nuopc/shr/shr_nuopc_methods_mod.F90 [hunk 4]
```diff
@@ -1317,7 +1318,7 @@ subroutine shr_nuopc_methods_FB_diagnose(FB, string, rc)
     integer                         :: i,j,n
     integer                         :: fieldCount, lrank
     character(ESMF_MAXSTR), pointer :: lfieldnamelist(:)
-    character(len=64)               :: lstring
+    character(len=128)              :: lstring
```
Comment (jedwards4b) [inline src/drivers/nuopc/shr/shr_nuopc_methods_mod.F90]: Shouldn't we use CL or CX here?

**Review by jedwards4b — APPROVED** [review]

File: src/drivers/nuopc/cime_driver/esm.F90 (hunk 1 above)
Comment (mvertens) [reply src/drivers/nuopc/cime_driver/esm.F90:324]: This was the original way. I have no idea why we want or do not want the pretty_print_nuopc. Frankly, I don't know what this produces. I am happy to take them out at this point since we are not using these lines.

File: src/drivers/nuopc/cime_driver/esm.F90 (hunk 2 above)
Comment (mvertens) [reply src/drivers/nuopc/cime_driver/esm.F90]: fixed.

File: src/drivers/nuopc/cime_driver/esm.F90 (hunk 3 above)
Comment (mvertens) [reply src/drivers/nuopc/cime_driver/esm.F90]: Sure. I tried to clean up as much as possible.

File: src/drivers/nuopc/shr/shr_nuopc_methods_mod.F90 (hunk 4 above)
Comment (mvertens) [reply src/drivers/nuopc/shr/shr_nuopc_methods_mod.F90]: Good point. I will change that to CS.

Comment (mvertens) [reply src/drivers/nuopc/shr/shr_nuopc_methods_mod.F90]: I fixed a bunch of other hard-wired character strings as well. Thanks so much for your quick review.

**Comment by rsdunlapiv** [comment]
```
qcmd -l walltime=4:00:00 --  ./create_test --xml-testlist ../src/drivers/nuopc/cime_config/testdefs/testlist_drv.xml --xml-machine cheyenne --xml-category nuopc --walltime 00:30:00 --compare jan24 --baseline-root /glade/scratch/dunlap/BASELINES/ --generate feb4
```

**Comment by rsdunlapiv** [comment]
Failures are as expected:

```
dunlap@cheyenne6:/glade/scratch/dunlap> ./cs.status.20190204_164304_aslv75 | grep "FAIL " | grep -v NLCOMP    FAIL ERR_Vnuopc_Ld5.f19_g16.BMOM.cheyenne_intel.allactive-nuopc_cap_io COMPARE_base_rest
    FAIL ERS_Vnuopc_Ld5.T62_g16.CMOM.cheyenne_intel.mom-nuopc_cap COMPARE_base_rest
    FAIL ERS_Vnuopc_Ld5.T62_g16.GMOM.cheyenne_intel.mom-nuopc_cap COMPARE_base_rest
    FAIL ERS_Vnuopc_Ln5.f19_g17.F2000Nuopc.cheyenne_intel.cam-nuopc_cap BASELINE jan24: DIFF
    FAIL ERS_Vnuopc_Ln5.f45_f45_mg37.I2000Clm50SpNuopc.cheyenne_intel.clm-nuopc_cap BASELINE jan24: DIFF
    FAIL ERS_Vnuopc_Ln9.f19_g17.X.cheyenne_intel BASELINE jan24: DIFF
    FAIL ERS_Vnuopc_Ln9_N3.f19_g17_rx1.A.cheyenne_intel TPUTCOMP Error: Computation time increase > 25 pct from baseline
    FAIL SMS_Vnuopc.f19_g17.X.cheyenne_intel BASELINE jan24: DIFF
    FAIL SMS_Vnuopc_Ld1_N3.f19_g17_rx1.A.cheyenne_intel TPUTCOMP Error: Computation time increase > 25 pct from baseline
    FAIL SMS_Vnuopc_Ld5.f19_g16.BMOM.cheyenne_intel.allactive-nuopc_cap_io MEMCOMP Error: Memory usage increase > 10% from baseline
    FAIL SMS_Vnuopc_Ln5.f19_f19_mg17.I2000Clm50SpGs.cheyenne_intel.clm-nuopc_cap BASELINE jan24: ERROR BFAIL baseline directory '/glade/scratch/dunlap/BASELINES/jan24/SMS_Vnuopc_Ln5.f19_f19_mg17.I2000Clm50SpGs.cheyenne_intel.clm-nuopc_cap' does not exist
    FAIL SMS_Vnuopc_Ln5.f19_g17.F2000Nuopc.cheyenne_intel.cam-nuopc_cap BASELINE jan24: DIFF
    FAIL SMS_Vnuopc_Ln5.f45_f45_mg37.I2000Clm50SpNuopc.cheyenne_intel.clm-nuopc_cap BASELINE jan24: DIFF
```
